test/parallel/test-permission-deny-fs-read.js

```diff
@@ -249,6 +249,21 @@ const gid = os.userInfo().gid;
   fs.open(regularFile, 'r', (err) => {
     assert.ifError(err);
   });
+
+  // Extra flags should not enable trivially bypassing all restrictions.
+  // See https://github.com/nodejs/node/issues/47090.
+  assert.throws(() => {
+    fs.openSync(blockedFile, fs.constants.O_RDONLY | fs.constants.O_NOCTTY);
```

Comment:
Does it cover string flags like: `a+`, `r`, 'w+', and o so on?

Reply:
IIRC the C++ layer only handles numeric flags and the JS layer translates strings accordingly, in which case the answer would be _yes_. If I'm misremembering and there are C++ APIs that accept flags as such strings, then the answer is likely _no_ and additional vulnerabilities in the existing permission model implementation might exist.